packages/flutter/test/widgets/scroll_activity_test.dart

```diff
@@ -19,7 +19,7 @@ void main() {
     final double thirty = controller.position.maxScrollExtent;
     controller.jumpTo(thirty);
     await tester.pump();
-    controller.jumpTo(thirty + 100.0); // past the end
+    controller.jumpTo(thirty + 200.0); // past the end
     await tester.pump();
     await tester.pumpWidget(MaterialApp(home: ListView(controller: controller, children: children(31))));
     expect(controller.position.pixels, thirty + 200.0); // same distance past the end
```

Comment:
Here, with old logic it was

* Scroll past the end by 100 pixels
* Add new item which maintains overscroll, therefore position receives additional 100 pixels + 100 pixels of ovescroll = 200

With new logic

* Scroll past the end
* Add new item which doesn't maintain ovescroll, therefore position doesn't receive additional 100 pixels. And we still want to keep it 200.0, because below we verify that we are still in the ovescroll. If we change the overscroll value to 100.0, the inserted item will make that there's now enough extent so we are no longer in overscroll.

Reply:
can you make a second copy of this test that's the same but instead of changing the jump, you change the expects?

Reply:
I'm now wondering whether this test could instead be entirely removed, since it tests the same logic as the new test that I added?

Reply:
having different variants of the test is good, especially for this subtle stuff, even if they theoretically test the same thing, because in practice they almost certainly don't exactly test the same thing.

Reply:
i still think it'd be good to have two versions of this test, one where we test the behaviour at thirty+100 and one thirty+200.

Reply:
Done